infra/testApi.mjs

```diff
@@ -1,82 +1,163 @@
+  })
+})).json())
 
-const mathjs = (await (await fetch('http://127.0.0.1:8000/api/mathjs', { 
-	method: 'POST', 
-	body: JSON.stringify({ formula: '(FPCore (x) (- (sqrt (+ x 1)) (sqrt x)))'})
-      })).json()).mathjs
+assert.equal(localError.tree['avg-error'] > 0, true)
 
-assert.equal(mathjs, "sqrt(x + 1.0) - sqrt(x)")
+// Alternatives endpoint
 
-const eval_sample = [[[1], -1.4142135623730951]]
-const exacts = (await (await fetch('http://127.0.0.1:8000/api/exacts', {method: 'POST', body: JSON.stringify({formula: "(FPCore (x) (- (sqrt (+ x 1))))", sample: eval_sample})})).json()).points
+const alternatives = (await (await fetch('http://127.0.0.1:8000/api/alternatives', {
+  method: 'POST', body: JSON.stringify({
+    formula: FPCoreFormula, sample: [[[
+      14.97651307489794
+    ], 0.12711304680349078]]
+  })
+})).json())  // HACK tiny sample
 
-assert.deepEqual(exacts, [[[1], -1.4142135623730951]])
+assert.equal(Array.isArray(alternatives.alternatives), true)
 
-const calculate = (await (await fetch('http://127.0.0.1:8000/api/calculate', {method: 'POST', body: JSON.stringify({formula: "(FPCore (x) (- (sqrt (+ x 1))))", sample: eval_sample})})).json()).points
+// Exacts endpoint
+const exacts = (await (await fetch('http://127.0.0.1:8000/api/exacts', {
+  method: 'POST', body: JSON.stringify({
+    formula: FPCoreFormula2, sample: eval_sample
+  })
+})).json()).points
 
 assert.deepEqual(exacts, [[[1], -1.4142135623730951]])
 
-const cost = (await (await fetch('http://127.0.0.1:8000/api/cost', {method: 'POST', body: JSON.stringify({formula: "(FPCore (x) (- (sqrt (+ x 1))))", sample: eval_sample})})).json())
+// Calculate endpoint
+const calculate = (await (await fetch('http://127.0.0.1:8000/api/calculate', {
+  method: 'POST', body: JSON.stringify({
+    formula: FPCoreFormula2, sample: eval_sample
+  })
+})).json()).points
 
-const sample3 = (await (await fetch('http://127.0.0.1:8000/api/sample', { method: 'POST', body: JSON.stringify({ formula: '(FPCore (x) (- (sqrt (+ x 1)) (sqrt x)))', seed: 5 }) })).json())
+assert.deepEqual(calculate, [[[1], -1.4142135623730951]])
 
-const localerror = (await (await fetch('http://127.0.0.1:8000/api/localerror', { method: 'POST', body: JSON.stringify({ formula: '(FPCore (x) (- (sqrt (+ x 1)) (sqrt x)))', sample: sample2.points }) })).json())
+// Cost endpoint
 
-assert.equal(localerror.tree['avg-error'] > 0, true)
+const cost = (await (await fetch('http://127.0.0.1:8000/api/cost', {
+  method: 'POST', body: JSON.stringify({
+    formula: FPCoreFormula2, sample: eval_sample
+  })
+})).json())
 assert.equal(cost.cost > 0, true)
 
+// MathJS endpoint
+
+const mathjs = (await (await fetch('http://127.0.0.1:8000/api/mathjs', {
+  method: 'POST',
+  body: JSON.stringify({ formula: FPCoreFormula })
+})).json()).mathjs
+
+assert.equal(mathjs, "sqrt(x + 1.0) - sqrt(x)")
+
+// Translate endpoint
+
 const languageList = ["python", "c", "fortran", "java", "julia", "matlab", "wls", "tex", "js"]
```

Comment:
If you combine this with the `expectedExpressions` below then both of these will be a lot more readable.